core/benches/scheduler.rs

```diff
@@ -0,0 +1,494 @@
+            scheduler::{PreLockFilterAction, Scheduler},
+            transaction_state::TransactionState,
+            transaction_state_container::{StateContainer, TransactionStateContainer},
+        },
+        TOTAL_BUFFERED_PACKETS,
+    },
+    solana_runtime_transaction::{
+        runtime_transaction::RuntimeTransaction, transaction_with_meta::TransactionWithMeta,
+    },
+    solana_sdk::{
+        compute_budget::ComputeBudgetInstruction,
+        hash::Hash,
+        instruction::Instruction,
+        message::Message,
+        pubkey::Pubkey,
+        signature::Keypair,
+        signer::Signer,
+        system_instruction,
+        transaction::{SanitizedTransaction, Transaction},
+    },
+    std::{
+        sync::{
+            atomic::{AtomicUsize, Ordering},
+            Arc,
+        },
+        time::{Duration, Instant},
+    },
+};
+
+#[global_allocator]
+static GLOBAL: Jemalloc = Jemalloc;
+
+// A non-contend low-prio tx, aka Tracer, is tag with this requested_loaded_accounts_data_size_limit
+const TAG_NUMBER: u32 = 1234;
+
+fn is_tracer<Tx: TransactionWithMeta + Send + Sync + 'static>(tx: &Tx) -> bool {
+    matches!(
+        tx.compute_budget_instruction_details()
+            .requested_loaded_accounts_data_size_limit(),
+        Some(TAG_NUMBER)
+    )
+}
+
+trait TestTxsBuilder {
+    // Tracer is a non-contend low-prio transfer transaction, it'd usually be inserted into the bottom
+    // of Container due to its low prio, but it should be scheduled early for better UX, if its reward
+    // is adequate from bolcker builder perspective.
+    fn build_tracer_transaction(&self) -> RuntimeTransaction<SanitizedTransaction> {
+        let payer = Keypair::new();
+        let to_pubkey = Pubkey::new_unique();
+        let mut ixs = vec![system_instruction::transfer(&payer.pubkey(), &to_pubkey, 1)];
+        ixs.push(ComputeBudgetInstruction::set_compute_unit_price(4));
+        ixs.push(ComputeBudgetInstruction::set_loaded_accounts_data_size_limit(TAG_NUMBER));
+        let message = Message::new(&ixs, Some(&payer.pubkey()));
+        let tx = Transaction::new(&[payer], message, Hash::default());
+        RuntimeTransaction::from_transaction_for_tests(tx)
+    }
+
+    fn build(
+        &self,
+        count: usize,
+        is_single_payer: bool,
+    ) -> Vec<RuntimeTransaction<SanitizedTransaction>> {
+        let mut transactions = Vec::with_capacity(count);
+        // non-contend low-prio tx is first received
+        transactions.push(self.build_tracer_transaction());
+
+        let compute_unit_price = 1_000;
+        let single_payer = Keypair::new();
+        for _n in 1..count {
+            let payer = if is_single_payer {
+                // recreate keypair from single_payer
+                Keypair::from_bytes(&single_payer.to_bytes()).expect("Failed to create Keypair")
+            } else {
+                Keypair::new()
+            };
```

Comment:
You can just do this from rust's temporary lifetime extension:
```suggestion
            let payer = if is_single_payer {
                &single_payer
            } else {
                &Keypair::new()
            };
```

[explanation here](https://users.rust-lang.org/t/how-to-understand-temporary-lifetime-extension-correctly/70503/2)

Reply:
TIL